src/cartservice/src/cartstore/RedisCartStore.cs

```diff
@@ -16,95 +16,18 @@
 using System.Linq;
 using System.Threading.Tasks;
 using Grpc.Core;
-using StackExchange.Redis;
+using Microsoft.Extensions.Caching.Distributed;
 using Google.Protobuf;
 
 namespace cartservice.cartstore
 {
     public class RedisCartStore : ICartStore
     {
-        private const string CART_FIELD_NAME = "cart";
-        private const int REDIS_RETRY_NUM = 30;
+        private readonly IDistributedCache _cache;
 
-        private volatile ConnectionMultiplexer redis;
-        private volatile bool isRedisConnectionOpened = false;
-
-        private readonly object locker = new object();
-        private readonly byte[] emptyCartBytes;
-        private readonly string connectionString;
-
-        private readonly ConfigurationOptions redisConnectionOptions;
-
-        public RedisCartStore(string redisAddress)
-        {
-            // Serialize empty cart into byte array.
-            var cart = new Hipstershop.Cart();
-            emptyCartBytes = cart.ToByteArray();
-            connectionString = $"{redisAddress},ssl=false,allowAdmin=true,abortConnect=false";
-
-            redisConnectionOptions = ConfigurationOptions.Parse(connectionString);
-
-            // Try to reconnect multiple times if the first retry fails.
-            redisConnectionOptions.ConnectRetry = REDIS_RETRY_NUM;
-            redisConnectionOptions.ReconnectRetryPolicy = new ExponentialRetry(1000);
-
-            redisConnectionOptions.KeepAlive = 180;
-        }
-
-        public Task InitializeAsync()
+        public RedisCartStore(IDistributedCache cache)
         {
-            EnsureRedisConnected();
-            return Task.CompletedTask;
-        }
-
-        private void EnsureRedisConnected()
```

Comment:
**Praise:**
I like that this new approach gets rid of the `EnsureRedisConnected()` method.
[The best code is no code](https://blog.codinghorror.com/the-best-code-is-no-code-at-all/).

Reply:
Good point, I just tested to log the differences while we are here.

So if I delete the `redis-cart` deployment and hit refresh on the `frontend`:

Before it was this error in the `cartservice`'s logs:
```
Error status code 'FailedPrecondition' with detail 'Can't access cart storage. System.ApplicationException: Wasn't able to connect to redis
         at cartservice.cartstore.RedisCartStore.EnsureRedisConnected() in /app/cartstore/RedisCartStore.cs:line 108
         at cartservice.cartstore.RedisCartStore.GetCartAsync(String userId) in /app/cartstore/RedisCartStore.cs:line 195' raised.
```

After and now it is:
```
Error status code = FailedPrecondition desc = Can't access cart storage. StackExchange.Redis.RedisConnectionException: No connection is active/available to service this operation: HMGET ae977b82-29ac-42ac-a1e2-9c1f932d5775; Connection reset by peer, mc: 1/1/0, mgr: 10 of 10 available, clientName: cartservice-546bfd77f4-b6gd9, IOCP: (Busy=0,Free=1000,Min=1,Max=1000), WORKER: (Busy=2,Free=32765,Min=1,Max=32767), v: 2.2.4.27433
 ---> StackExchange.Redis.RedisConnectionException: SocketFailure (ReadSocketError/ConnectionReset, 0-read, last-recv: 0) on redis-cart:6379/Subscription, Flushed/Faulted, last: SUBSCRIBE, origin: ReadFromPipe, outstanding: 3, last-read: 0s ago, last-write: 0s ago, keep-alive: 60s, state: ConnectedEstablishing, mgr: 6 of 10 available, last-heartbeat: never, last-mbeat: 0s ago, global: 0s ago, v: 2.2.4.27433
 ---> Pipelines.Sockets.Unofficial.ConnectionResetException: Connection reset by peer
 ---> System.Net.Sockets.SocketException (104): Connection reset by peer
   at Pipelines.Sockets.Unofficial.Internal.Throw.Socket(Int32 errorCode) in /_/src/Pipelines.Sockets.Unofficial/Internal/Throw.cs:line 59
   at Pipelines.Sockets.Unofficial.SocketAwaitableEventArgs.GetResult() in /_/src/Pipelines.Sockets.Unofficial/SocketAwaitableEventArgs.cs:line 74
   at Pipelines.Sockets.Unofficial.SocketConnection.DoReceiveAsync() in /_/src/Pipelines.Sockets.Unofficial/SocketConnection.Receive.cs:line 64
   --- End of inner exception stack trace ---
   at System.IO.Pipelines.Pipe.GetReadResult(ReadResult& )
   at System.IO.Pipelines.Pipe.GetReadAsyncResult()
   at StackExchange.Redis.PhysicalConnection.ReadFromPipe() in /_/src/StackExchange.Redis/PhysicalConnection.cs:line 1496
   --- End of inner exception stack trace ---
   --- End of inner exception stack trace ---
   at StackExchange.Redis.ConnectionMultiplexer.ThrowFailed[T](TaskCompletionSource`1 source, Exception unthrownException) in /_/src/StackExchange.Redis/ConnectionMultiplexer.cs:line 2760
--- End of stack trace from previous location ---
   at Microsoft.Extensions.Caching.StackExchangeRedis.RedisExtensions.HashMemberGetAsync(IDatabase cache, String key, String[] members)
   at Microsoft.Extensions.Caching.StackExchangeRedis.RedisCache.GetAndRefreshAsync(String key, Boolean getData, CancellationToken token)
   at Microsoft.Extensions.Caching.StackExchangeRedis.RedisCache.GetAsync(String key, CancellationToken token)
   at cartservice.cartstore.RedisCartStore.GetCartAsync(String userId) in /app/cartstore/RedisCartStore.cs:line 104
```